tests/acceleration/test_acceleration_framework.py

```diff
@@ -0,0 +1,455 @@
+    plugin.
+    """
+    with tempfile.TemporaryDirectory() as tempdir:
+        TRAIN_KWARGS = {
+            **bad_train_kwargs,
+            "output_dir": tempdir,
+        }
+        model_args, data_args, training_args, tune_config = causal_lm_train_kwargs(
+            TRAIN_KWARGS
+        )
+        quantized_lora_config = QuantizedLoraConfig(auto_gptq=AutoGPTQLoraConfig())
+
+        # 1. activate the accelerated peft plugin
+        # 2. demonstrate that the invalid arguments will be checked
+        with pytest.raises(exception, match=exception_msg):
+            sft_trainer.train(
+                model_args,
+                data_args,
+                training_args,
+                tune_config,
+                quantized_lora_config=quantized_lora_config,
+            )
+
+
+if is_fms_accelerate_available(plugins="peft"):
+    acceleration_configs_map = [
+        (
+            QuantizedLoraConfig(bnb_qlora=BNBQLoraConfig()),
+            "TinyLlama/TinyLlama-1.1B-Chat-v1.0",
+            (
+                "peft.quantization.bitsandbytes",
+                create_mock_plugin_class_and_spy("PluginMock", BNBAccelerationPlugin),
+            ),
+        ),
+        (
+            QuantizedLoraConfig(auto_gptq=AutoGPTQLoraConfig()),
+            "TheBloke/TinyLlama-1.1B-Chat-v0.3-GPTQ",
+            (
+                "peft.quantization.auto_gptq",
+                create_mock_plugin_class_and_spy(
+                    "PluginMock", AutoGPTQAccelerationPlugin
+                ),
+            ),
+        ),
+    ]
+
+
+@pytest.mark.skipif(
+    not is_fms_accelerate_available(plugins="peft"),
+    reason="Only runs if fms-accelerate is installed along with accelerated-peft plugin",
+)
+@pytest.mark.parametrize(
+    "quantized_lora_config,model_name_or_path,mock_and_spy",
+    acceleration_configs_map,
+    ids=["bitsandbytes", "auto_gptq"],
+)
+def test_framework_intialized_properly_peft(
+    quantized_lora_config, model_name_or_path, mock_and_spy
+):
+    """Ensure that specifying a properly configured acceleration dataclass
+    properly activates the framework plugin and runs the train sucessfully.
+    """
+    with tempfile.TemporaryDirectory() as tempdir:
+        TRAIN_KWARGS = {
+            **BASE_LORA_KWARGS,
+            **{
+                "fp16": True,
+                "model_name_or_path": model_name_or_path,
+                "output_dir": tempdir,
+                "save_strategy": "no",
+            },
+        }
+        model_args, data_args, training_args, tune_config = causal_lm_train_kwargs(
+            TRAIN_KWARGS
+        )
+        installation_path, (MockedPlugin, spy) = mock_and_spy
+
+        # 1. mock a plugin class
+        # 2. register the mocked plugin
+        # 3. call sft_trainer.train
+        with build_framework_and_maybe_instantiate(
+            [([installation_path], MockedPlugin)],
+            instantiate=False,
+        ):
+            sft_trainer.train(
```

Comment:
also will produce artifacts correct? to avoid artifacts getting generated and stored in local machines whenever we run tests locally, please use tempdir `    with tempfile.TemporaryDirectory() as tempdir:` whenever applicable. 

This will ensure directory is cleaned after test is complete. 
See test https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/tests/test_sft_trainer.py#L351C1-L351C51